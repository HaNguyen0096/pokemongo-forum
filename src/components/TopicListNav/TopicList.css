.topicNav{
  display: block;
  padding: 0;
}

.topicNameBar{

  display: inline-block;
  font-weight: bold;
  vertical-align: bottom;
}


.topicNames{
  text-decoration: none;
}


.topicNames:link{
  color: black;
}

.topicNames:visited{
  color: black;
}

.topicNames:hover{
  color: rgb(23,156,106);
}

.countThread{
  color: rgb(70, 70, 70);
}

hr{
  border: 0;
  height: 1px;
  background-color: rgb(23,156,106);
  width: 90%;
}


