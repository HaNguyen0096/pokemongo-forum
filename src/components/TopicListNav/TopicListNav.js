import React, { Component } from 'react'
import { Link } from 'react-router-dom'

export default class TopicListNav extends Component{

  render(){
    const { topic } = this.props
    let count = this.props.countThread
    return (
      <div className='TopicNav'>
      <nav role="navigation">
        <ul className='TopicNav_List'>
          <Link to={`/topic/${topic.id}`} className='TopicListName'>
            <h2 className='TopicListItem__heading'>{topic.topic_name}</h2>
          </Link>
          <p>{topic.topic_content}</p>
          <p>Threads: {count}</p>
        </ul>
      </nav>
      </div>
    )
  }
}