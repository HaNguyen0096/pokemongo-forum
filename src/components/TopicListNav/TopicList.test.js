import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import TopicListNav from './TopicListNav'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

const testTopic = [
  {
    id: 1,
    topic_name: 'General',
    topic_content: 'General discussion about Pokemon Go!.'
  },
]

describe('TopicList component', () => {

  it('renders TopicList without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>
                      <TopicListNav topic={testTopic}/>
                    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI for TopicList as expected', () => {
    const tree = renderer
      .create(<BrowserRouter>
                <TopicListNav topic={testTopic} />
              </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('renders TopicList by default', () => {
    const wrapper = shallow(<TopicListNav topic={testTopic} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });

})