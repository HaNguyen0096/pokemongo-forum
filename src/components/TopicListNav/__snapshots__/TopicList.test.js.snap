// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopicList component renders TopicList by default 1`] = `
<div
  className="topicNav"
>
  <nav
    role="navigation"
  >
    <ul
      className="topicNav_List"
    >
      <div
        className="topicNameBar"
      >
        <Link
          className="topicNames"
          to="/topic/undefined"
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon="folder"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          />
           
        </Link>
      </div>
      <p
        className="topicDescription"
      />
      <p
        className="countThread"
      >
        Threads: 
      </p>
    </ul>
  </nav>
  <hr />
</div>
`;

exports[`TopicList component renders the UI for TopicList as expected 1`] = `
<div
  className="topicNav"
>
  <nav
    role="navigation"
  >
    <ul
      className="topicNav_List"
    >
      <div
        className="topicNameBar"
      >
        <a
          className="topicNames"
          href="/topic/undefined"
          onClick={[Function]}
        >
           
        </a>
      </div>
      <p
        className="topicDescription"
      />
      <p
        className="countThread"
      >
        Threads: 
      </p>
    </ul>
  </nav>
  <hr />
</div>
`;
